packages/remix-react/transition.ts

```diff
@@ -596,6 +600,28 @@ export function createTransitionManager(init: TransitionManagerInit) {
     }
   }
 
+  function isIndexRequestUrl(url: URL) {
+    let indexRequest = false;
+
+    for (let param of url.searchParams.getAll("index")) {
+      if (!param) {
+        indexRequest = true;
+      }
+    }
+
+    return indexRequest;
```

Comment:
```suggestion
    for (let param of url.searchParams.getAll("index")) {
      if (!param) return true;
    }

    return false;
```

Reply:
Okay, this is weird I know ... but folks may need to use `?index=123` for bookmark pagination (or really anything), so the way we can do this without clobbering the `index` search param namespace is *only a bare index param means the index route*.

```
✅ /foo?index
✅ /foo?index&index=123
✅ /foo?index=123&index
❌ /foo?index=123
```

This code does exactly that, but it's not super clear since it's relying on "falsey" empty string and the fact that the loop happened at all. I'd go with:


```js
for (let param of url.searchParams.getAll("index")) {
  if (param === "") return true;
}
```